gslb/test/integration/third_party_vips/gslb_hostrule_test.go

```diff
@@ -875,3 +875,55 @@ func TestGSLBHostRuleHmRefToHealthMonitorTemplate(t *testing.T) {
 	// the GS should contain the hm template from the GSLB hostrule.
 	verifyMembers(nil, &gslbHrHmTemplate)
 }
+
+func TestGSLBHostRuleWithPublicIPUpdate(t *testing.T) {
+	testPrefix := "gdp-gslbhr-algo-"
+	gslbHRName := "test-gslb-hr"
+	hmRefs := []string{"my-hm1"}
+	sp := "gap-1"
+	ttl := 10
+	gdpPa := gslbalphav1.PoolAlgorithmSettings{
+		LBAlgorithm: "GSLB_ALGORITHM_ROUND_ROBIN",
+	}
+	addTestGDPWithProperties(t, hmRefs, nil, &ttl, &sp, &gdpPa)
+	ingObj, routeObj := addIngressAndRouteObjects(t, testPrefix)
+	verifyMembers := func(pa []gslbalphav1.PublicIPElem) {
+		var expectedMembers []nodes.AviGSK8sObj
+		expectedMembers = append(expectedMembers, getTestGSMemberFromIng(t, ingObj, ingCluster, 1, 10))
+		expectedMembers = append(expectedMembers, getTestGSMemberFromRoute(t, routeObj, routeCluster, 1, 10))
+		if pa != nil {
+			expectedMembers[0].PublicIP = pa[0].IP
+			expectedMembers[1].PublicIP = pa[1].IP
+		}
+		g := gomega.NewGomegaWithT(t)
+
+		g.Eventually(func() bool {
+			return verifyGSMembers(t, expectedMembers, routeObj.Spec.Host, utils.ADMIN_NS, hmRefs, nil, &sp,
+				&ttl, nil, defaultPath, TlsTrue, nil)
+		}, 5*time.Second, 1*time.Second).Should(gomega.Equal(true))
+	}
+	hostName := routeObj.Spec.Host
+	oldObj := addGSLBHostRule(t, gslbHRName, gslbutils.AVISystem, hostName, hmRefs, nil, nil, &ttl,
+		ingestion.GslbHostRuleAccepted, "")
+	gslbhostrulepip := []gslbalphav1.PublicIPElem{
+		{Cluster: K8sContext, IP: "10.20.30.40"},
+		{Cluster: OshiftContext, IP: "10.20.40.50"}}
+	newObj := getGSLBHostRule(t, oldObj.Name, oldObj.Namespace)
+	newObj.Spec.PublicIP = gslbhostrulepip
```

Comment:
This seems to be wrong. We have to assign the IP to `thirdPartyMembers`.

Reply:
added another Ut to test third party